### PR TITLE
Bump boltons to 24.1.*

### PR DESCRIPTION
**Pull request by yangdanny97**
Manually update stubs to reflect changes

Closes https://github.com/python/typeshed/pull/12944

### CODE REVIEW

File: stdlib/typing.pyi [hunk 1]
```diff
@@ -352,7 +352,7 @@ def type_check_only(func_or_cls: _F) -> _F: ...
 
 class _Alias:
     # Class for defining generic aliases for library types.
-    def __getitem__(self, typeargs: Any) -> Any: ...
+    def __getitem__(self, typeargs: Any) -> object: ...
```
Comment (AlexWaygood) [inline stdlib/typing.pyi:355]: Hmm, what's the rationale behind this change? It seems somewhat unrelated to the other changes in this PR

Comment (yangdanny97) [reply stdlib/typing.pyi:355]: oops, I think I accidentally unstashed something.

**Review by AlexWaygood — APPROVED** [review]
Thanks!
